src/components/annotations/draw_arrow_head.js

```diff
@@ -79,41 +103,35 @@ module.exports = function drawArrowHead(el3, ends, options) {
             // combine the two
             dashArray = '';
 
-        if(pathlen < backOff) {
+        if(pathlen < backOff || pathlen < startBackOff || pathlen < backOff + startBackOff) {
             hideLine();
             return;
         }
 
-        if(doStart) {
-            var start0 = el.getPointAtLength(0);
-            var dstart = el.getPointAtLength(0.1);
 
-            startRot = Math.atan2(start0.y - dstart.y, start0.x - dstart.x);
-            start = el.getPointAtLength(Math.min(backOff, pathlen));
+        var start0 = el.getPointAtLength(0);
+        var dstart = el.getPointAtLength(0.1);
 
-            if(backOff) dashArray = '0px,' + backOff + 'px,';
-        }
+        startRot = Math.atan2(start0.y - dstart.y, start0.x - dstart.x);
+        start = el.getPointAtLength(Math.min(startBackOff, pathlen));
 
-        if(doEnd) {
-            var end0 = el.getPointAtLength(pathlen);
-            var dend = el.getPointAtLength(pathlen - 0.1);
+        dashArray = '0px,' + startBackOff + 'px,';
 
-            endRot = Math.atan2(end0.y - dend.y, end0.x - dend.x);
-            end = el.getPointAtLength(Math.max(0, pathlen - backOff));
+        var end0 = el.getPointAtLength(pathlen);
+        var dend = el.getPointAtLength(pathlen - 0.1);
 
-            if(backOff) {
-                var shortening = dashArray ? 2 * backOff : backOff;
-                dashArray += (pathlen - shortening) + 'px,' + pathlen + 'px';
-            }
-        }
-        else if(dashArray) dashArray += pathlen + 'px';
+        endRot = Math.atan2(end0.y - dend.y, end0.x - dend.x);
+        end = el.getPointAtLength(Math.max(0, pathlen - backOff));
+
+        var shortening = dashArray ? startBackOff + backOff : backOff;
+        dashArray += (pathlen - shortening) + 'px,' + pathlen + 'px';
 
-        if(dashArray) el3.style('stroke-dasharray', dashArray);
+        el3.style('stroke-dasharray', dashArray);
     }
 
     function hideLine() { el3.style('stroke-dasharray', '0px,100px'); }
 
-    function drawhead(p, rot) {
+    function drawhead(headStyle, p, rot, scale) {
```

Comment:
This will work, but it's a bit confusing to reuse the names from the outer scope - perhaps rename to something like `thisHeadStyle` and `thisScale`?

Reply:
Changed names to `arrowHeadStyle` and `arrowScale`